beacon-chain/execution/service.go

```diff
@@ -816,9 +850,25 @@ func (s *Service) ensureValidPowchainData(ctx context.Context) error {
 			CurrentEth1Data:   s.latestEth1Data,
 			ChainstartData:    s.chainStartData,
 			BeaconState:       pbState,
-			Trie:              s.depositTrie.ToProto(),
 			DepositContainers: s.cfg.depositCache.AllDepositContainers(ctx),
 		}
+		if features.Get().EnableEIP4881 {
+			trie, ok := s.depositTrie.(*depositsnapshot.DepositTree)
+			if !ok {
+				return errors.New("deposit trie was not EIP4881 DepositTree")
+			}
+			snapshot, err := trie.GetSnapshot()
+			if err != nil {
+				return err
+			}
+			eth1Data.DepositSnapshot = snapshot.ToProto()
+		} else {
+			trie, ok := s.depositTrie.(*trie.SparseMerkleTrie)
+			if !ok {
+				return errors.New("deposit trie was not SparseMerkleTrie")
+			}
+			eth1Data.Trie = trie.ToProto()
+		}
```

Comment:
This branching could be avoided if you added `ToProto` to depositsnapshot.DepositTree

Reply:
I think we will still have to branch since they are different proto objects, however I have simplified the logic. Let me know if it works